src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,90 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (movie: Movie) => void,
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isButtonDisabled = Boolean(title && imgUrl && imdbUrl && imdbId);
```

Comment:
```suggestion
  const isButtonDisabled = title && imgUrl && imdbUrl && imdbId;
```
it already has boolean type